### PR TITLE
Move all validation to create.go and add more unit testing.

### PR DESCRIPTION
**Pull request by G-Harmon**
-Creates an interface for Validations and a fake Validator for unit tests.
-Adds a unit test for create.go:createIngressAndLoadBalancer.

cc @nikhiljindal @csbell

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/k8s-multicluster-ingress/183)

### CODE REVIEW

File: app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go [hunk 1]
```diff
@@ -100,8 +100,8 @@ func TestCreateLoadBalancer(t *testing.T) {
 		//   1x GET-Service for path=/foo.
 		// The last action should always be GET-Ingress, once for each cluster.
 		// This is how we get 3 actions for 1 cluster and 5 actions for the other.
-		fetchedSvc = fetchedSvc || len(actions) == 5
```
Comment (nikhiljindal) [inline app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go]: You'll need to update the above comment as well

File: app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go [hunk 2]
```diff
@@ -616,10 +579,8 @@ func verifyRemoveClustersResult(lbc *Syncer, expectedClusters, expectedIGlinks [
 		return fmt.Errorf("unexpected IG links on firewall rule, expected: %v, got: %v", expectedFWIGLinks, fw.IGLinks)
 	}
 	// Verify that the load balancer is spread to both clusters.
-	if err := verifyClusters(lbc, expectedClusters); err != nil {
-		return err
-	}
-	return nil
+	err := verifyClusters(lbc, expectedClusters)
```
Comment (nikhiljindal) [inline app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go]: I personally prefer code 1 over code 2.
code 1:
```go
if err := doStep1(); err != nil {
  return err
}
if err := doStep2(); err != nil {
  return err
}
return nil
```
code 2:
```go
if err := doStep1(); err != nil {
  return err
}
return doStep2()
```
While code 2 is less lines, code 1 allows for easy extension (can add/remove/reorder steps easily).

If you do want to switch this to code 2, you can combine both the lines to return verifyClusters directly

File: app/kubemci/pkg/ingress/ingresssyncer.go [hunk 3]
```diff
@@ -163,8 +163,6 @@ func unmarshall(filename string, ing *v1beta1.Ingress) error {
 		return err
 	}
 	// Unmarshall into ingress struct.
-	if err := yaml.Unmarshal(bytes, ing); err != nil {
-		return err
-	}
-	return nil
+	err = yaml.Unmarshal(bytes, ing)
```
Comment (nikhiljindal) [inline app/kubemci/pkg/ingress/ingresssyncer.go]: same as above

File: app/kubemci/pkg/validations/validations.go [hunk 4]
```diff
@@ -27,8 +27,22 @@ import (
 	"k8s.io/api/extensions/v1beta1"
 )
 
+// Validator is a concrete implementation of ValidatorInterface.
+type Validator struct {
```
Comment (nikhiljindal) [inline app/kubemci/pkg/validations/validations.go]: This can be private?

File: app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go (hunk 2 above)
Comment (G-Harmon) [reply app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go]: This was a lint error when it was "code 1", which was surprising!

Comment (nikhiljindal) [reply app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer_test.go]: I just tried lint on HEAD and didnt get any error?

File: app/kubemci/pkg/ingress/ingresssyncer.go (hunk 3 above)
Comment (G-Harmon) [reply app/kubemci/pkg/ingress/ingresssyncer.go]: reverted!

File: app/kubemci/pkg/validations/validations.go (hunk 4 above)
Comment (G-Harmon) [reply app/kubemci/pkg/validations/validations.go]: done. I hadn't thought of that.

File: app/kubemci/pkg/validations/validations.go [hunk 5]
```diff
@@ -27,8 +27,22 @@ import (
 	"k8s.io/api/extensions/v1beta1"
 )
 
+// Validator is a concrete implementation of ValidatorInterface.
```
Comment (nikhiljindal) [inline app/kubemci/pkg/validations/validations.go]: small v now

File: app/kubemci/pkg/validations/validations.go [hunk 6]
```diff
@@ -95,7 +109,7 @@ func nodePortSameInAllClusters(backend v1beta1.IngressBackend, namespace string,
 }
 
 // ServerVersionsNewEnough returns an error if the version of any cluster is not supported.
```
Comment (nikhiljindal) [inline app/kubemci/pkg/validations/validations.go]: same small s now

Comment (G-Harmon) [reply app/kubemci/pkg/validations/validations.go]: done.

File: app/kubemci/pkg/validations/validations.go (hunk 5 above)
Comment (G-Harmon) [reply app/kubemci/pkg/validations/validations.go]: Done.

**Pull request merged**